### PR TITLE
[BREAKING CHANGE] Cleanup TraceId/SpanId classes. Ensure consistency.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Summary of changes:
* getHexLength renamed to getLength
* bytesFromHex/bytesToHex renamed to asBytes/fromBytes.
* bytesFromHex/bytesToHex renamed to asLongHigh/asLongLow
* Update javadoc.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Fixes https://github.com/open-telemetry/opentelemetry-java/issues/2666

### CODE REVIEW

File: api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java [hunk 1]
```diff
@@ -44,84 +58,107 @@ public static String getInvalid() {
...
-  public static long traceIdLowBytesAsLong(CharSequence traceId) {
+  /**
+   * Returns the "low part" of the {@code long} values representation of the {@code TraceId}
+   * converted from the given lowercase hex (base16) representation.
+   *
+   * @param traceId the lowercase hex (base16) representation of the {@code TraceId}.
+   * @return the {@code long} value representation of the {@code TraceId}.
+   * @throws NullPointerException if {@code traceId} is null.
+   */
+  public static long asLongLow(CharSequence traceId) {
```
Comment (Oberon00) [inline api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: Using `as` for longLow/High is misleading IMHO. This is not the trace ID "as" a longLow. It is only a part of the trace ID. Maybe `getLongLow`  would be better.

Comment (jkwatson) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: `getLowAsLong()` is accurate, but might be a bit wordy?

Comment (jkwatson) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: `lowAsLong()` ?

**Comment by jkwatson** [comment]
aside from the tricky trace id low/high as longs, this looks good to me.

File: api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java (hunk 1 above)
Comment (bogdandrutu) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: I would try to make an exercise and write them in code:

```java
TraceId.asLongLow("12345678123456781234567812345678")
TraceId.getLongLow("12345678123456781234567812345678")
TraceId.getLowAsLong("12345678123456781234567812345678")
TraceId.lowAsLong("12345678123456781234567812345678")
TraceId.asLongLowPart("12345678123456781234567812345678")
TraceId.lowPartAsLong("12345678123456781234567812345678")
```

```java
SpanId.asLong("1234567812345678")
```

So far these were the suggestions, any recommendation based on this?

Comment (bogdandrutu) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: For consistency reason I kind of like `asLongLowPart` or `asLongLow`, but I also see a bit the fact that it is not converting the entire id, only extracts one of the parts. But on the other hand the returned value represents the LowPart of the id

Comment (jkwatson) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: I like `lowPartAsLong()` the best of those options.

Comment (bogdandrutu) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java:154]: Now that I read it I feel the same. Sounds more like English and also describes what it happens

Comment (bogdandrutu) [reply api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java]: Changed 👍

**Review by jkwatson — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]
